### ISSUE TITLE
Unable to start debugging on .go files due to path error

### ISSUE THREAD

**Issue by Festum** [Microsoft/vscode-go#878]
Only happened on Golang files both portable version and install version. Debugging on another language(like Python) works well.
msg:
```
The program "${file}" does not exist: Error: ENOENT: no such file or directory, lstat 'C:\fe0\bin\VSCode\App\VSCode\${file}'
```
vs code version: 1.10.2
go version: 1.8

Any ideas?

**Comment by ramya-rao-a** [comment]
Can you share your debug configurations from `launch.json` ?

**Comment by ramya-rao-a** [comment]
Can you share the complete error and your `launch.json`?

Looks like the error is coming from https://github.com/Microsoft/vscode-go/blob/c530ed4c7cae1038df630307ff9a5eb173714e53/src/debugAdapter/goDebug.ts#L206

**Comment by DT-ZhangJia** [comment]
(VSCode) Try opening the folder where the .go file exists in. Don't open the .go file independently. Or VSCode could not create/read the .vscode folder nor the debug file.

**Comment by joaomoreno** [comment]
@bviktor has the same problem:

> - VSCode Version: 1.11.2
> - OS Version: 1703 15063.138
> 
> Steps to Reproduce:
> 
> 1) Get a VM image for win10 15.15063 from [modern.ie](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/)
> 
> 2) start it up, install updates, restart
> 
> 3) install vs code with defaults
> 
> 4) install [go runtime ](https://golang.org/dl/) with defaults
> 
> 5) start vs code
> 
> 6) install go extension, restart vs code
> 
> 7) create a new .go file:
> 
> ~~~
> package main
> import "fmt"
> func main() {
	> fmt.Println("foobar")
> }
> ~~~
> 
> 8) ctrl+f5
> 
> Expected output: `foobar` printed
> 
> Actual output: 
> 
> ~~~
> The program "${file}" does not exist: Error: ENOENT: no such file or directory, lstat 'C:\Program Files (x86)\Microsoft VS Code\${file}'
> ~~~
> 
> On the top the following appears:
> 
> ~~~
> Error: Failed to continue: "The program attribute must point to valid directory, .go file or executable."
> ~~~
> 
> Which you can either "close" or "open launch.json" but if you try the latter, nothing happens at all.

**Comment by ramya-rao-a** [comment]
@bviktor Can you share your debug configuration from the launch.json file?

**Comment by gaocuo** [comment]
Can not found any launch.json in the work dir and whole disk.

**Comment by scymen** [comment]
Hit the save error. 
Does anyone can tell me where is the file launch.json?

**Comment by jiaoyk** [comment]
same issue. No launch.json could be found on disk.

**Comment by ramya-rao-a** [comment]
The `launch.json` file would be in the `.vscode` folder in your workspace.
It gets created when you click on the gear icon in the debug view
If it doesn't exist, then VS Code tries to use some default configuration and tries to debug the current file that is open in the active editor.

This default configuration uses the `${file}` variable which should resolve to the current file that is open in the active editor.

It appears that `${file}` doesn't resolve properly in your cases.

I cannot repro it though.

Can one of you share a sample simple repo where this happens?
And share details like 
- which folder in this repo have you opened in VS Code?
- which file was open when you triggered debug?
- how did you trigger debug?

**Comment by bviktor** [comment]
@ramya-rao-a the opening post provides a step-by-step guide to reproduce the issue.

**Comment by bviktor** [comment]
Actually, it's the opening post of https://github.com/Microsoft/vscode/issues/25138

**Comment by bviktor** [comment]
- VSCode Version: 1.11.2
- OS Version: 1703 15063.138

Steps to Reproduce:

1) Get a VM image for win10 15.15063 from [modern.ie](https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/)

2) start it up, install updates, restart

3) install vs code with defaults

4) install [go runtime ](https://golang.org/dl/) with defaults

5) start vs code

6) install go extension, restart vs code

7) create a new .go file:

~~~
package main
import "fmt"
func main() {
	fmt.Println("foobar")
}
~~~

8) ctrl+f5

Expected output: `foobar` printed

Actual output: 

~~~
The program "${file}" does not exist: Error: ENOENT: no such file or directory, lstat 'C:\Program Files (x86)\Microsoft VS Code\${file}'
~~~

On the top the following appears:

~~~
Error: Failed to continue: "The program attribute must point to valid directory, .go file or executable."
~~~

Which you can either "close" or "open launch.json" but if you try the latter, nothing happens at all.

**Comment by ramya-rao-a** [comment]
Thanks @bviktor 

I can repro now.
The case is when there is no workspace open in VS Code but only a Go file is.

**Comment by ramya-rao-a** [comment]
This will be fixed by https://github.com/Microsoft/vscode-go/commit/42646afc2d2442b5e962d3125a7cbf61b98b2a0a

And the fix will be out in the next update

**Comment by bviktor** [comment]
Thanks a bunch!

**Comment by ramya-rao-a** [comment]
The fix for this is out in the latest update (0.6.60)

**Comment by bviktor** [comment]
Confirmed to be working now, thanks again.
